pkg/trace/stats/statsraw_test.go

```diff
@@ -84,15 +86,110 @@ func TestGrainWithPeerService(t *testing.T) {
 	})
 }
 
-func TestGrainWithExtraTags(t *testing.T) {
+func TestGrainWithPeerTags(t *testing.T) {
+	t.Run("none present", func(t *testing.T) {
+		assert := assert.New(t)
+		s := pb.Span{
+			Service:  "thing",
+			Name:     "other",
+			Resource: "yo",
+			Meta:     map[string]string{"span.kind": "client", "peer.service": "aws-s3"},
+		}
+		aggr, et := NewAggregationFromSpan(&s, "", PayloadAggregationKey{
+			Env:         "default",
+			Hostname:    "default",
+			ContainerID: "cid",
+		}, true, []string{"aws.s3.bucket", "db.instance", "db.system"})
+
+		assert.Equal(Aggregation{
+			PayloadAggregationKey: PayloadAggregationKey{
+				Env:         "default",
+				Hostname:    "default",
+				ContainerID: "cid",
+			},
+			BucketsAggregationKey: BucketsAggregationKey{
+				Service:     "thing",
+				SpanKind:    "client",
+				Name:        "other",
+				Resource:    "yo",
+				PeerService: "aws-s3",
+			},
+		}, aggr)
+		assert.Nil(et)
+	})
+	t.Run("partially present", func(t *testing.T) {
+		assert := assert.New(t)
+		s := pb.Span{
+			Service:  "thing",
+			Name:     "other",
+			Resource: "yo",
+			Meta:     map[string]string{"span.kind": "client", "peer.service": "aws-s3", "aws.s3.bucket": "bucket-a"},
+		}
+		aggr, et := NewAggregationFromSpan(&s, "", PayloadAggregationKey{
+			Env:         "default",
+			Hostname:    "default",
+			ContainerID: "cid",
+		}, true, []string{"aws.s3.bucket", "db.instance", "db.system"})
+
+		assert.Equal(Aggregation{
+			PayloadAggregationKey: PayloadAggregationKey{
+				Env:         "default",
+				Hostname:    "default",
+				ContainerID: "cid",
+			},
+			BucketsAggregationKey: BucketsAggregationKey{
+				Service:      "thing",
+				SpanKind:     "client",
+				Name:         "other",
+				Resource:     "yo",
+				PeerService:  "aws-s3",
+				PeerTagsHash: 6956349601612342508,
```

Comment:
Not a huge fan of hard-coding the values. I could make a func that does the fnv hash but that's pretty close already to the underlying func.